### ISSUE TITLE
jsc: complete notification (ostate: 9, nstate: 11) fires before exit_status values are available

### ISSUE THREAD

**Issue by trws** [flux-framework/flux-core#915]
See subject.  I'll circle back around with details if I find them.

**Comment by dongahn** [comment]
@trws: I started to look at this issue and I realized I need more details to make progress. There could be a bug in JSC itself, but it is hard to reason about this without a test case. 

My findings are:

1) exit code aggregation should be on by default, and its eventing flow is such that this condition shouldn't happen at the wreck level. More specifically, `wrexecd` installs exitstatus_watcher, which ultimately stops the reactor upon which the `complete` event is generated. So  there is a clear synchronization between exit_code and the `complete` event. JSC would simply have relayed this.  

2) Under no aggregation, it is true that a commit is omitted after each exit_code is being put in kvs. (unless commit on exit option is given). But @grondo is telling me that ultimately there is a commit right after state change, so this condition shouldn't happen either under this mode...
